k8s/object.bzl

```diff
@@ -159,11 +162,18 @@ def _common_impl(ctx):
     user_arg = "$(cat %s)" % _runfiles(ctx, user_file)
     files += [user_file]
 
+  namespace_file = None
+
   namespace_arg = ctx.attr.namespace
   namespace_arg = ctx.expand_make_variables("namespace", namespace_arg, {})
   if "{" in ctx.attr.namespace:
     namespace_file = ctx.new_file(ctx.label.name + ".namespace-name")
     _resolve(ctx, ctx.attr.namespace, namespace_file)
+
+  if ctx.file.namespace_file:
+    namespace_file = ctx.file.namespace_file
```

Comment:
If you dont add the namespace_file to the files, will it still be available in the struct returned below in line 210? If not, then this will result in failed builds for users that run with sandboxing enabled

Reply:
> If you dont add the namespace_file to the files, will it still be available in the struct returned below in line 210? If not, then this will result in failed builds for users that run with sandboxing enabled

This file also generated above
```
  if "{" in ctx.attr.namespace:
    namespace_file = ctx.new_file(ctx.label.name + ".namespace-name")
    _resolve(ctx, ctx.attr.namespace, namespace_file)
```

I just extended the original behavior: upstream version allows to specify namespace name as constant or use stamped variables. Now you can specify name based on file content which generated by bazel

Reply:
Fixed